### ISSUE TITLE
Refactor search [Phase II]

### ISSUE THREAD

**Issue by hrishikeshpaul** [hubmapconsortium/ccf-asct-reporter#96]
Tasks:
- [ ] Below the search-bar 3 checkboxes need to be added. Anatomical structures, Cell Types and Biomarkers. Selecting these checkboxes should show only those entities. Eg. if you select AS and CT, only entities that are AS and CT should be shown. Think of this as a filter.
- [ ] On clicking on a name (selecting it) the page should scroll to that location on the visualization. This will require you to use the `x` and `y` coordinates of that node, that you might have to retrieve from the store (I'm not sure if the data for the search nodes have it).
- [ ]  A "deselect all" button to quickly clear all selected structures.
- [ ] Update documentation

**Comment by macvogelsang** [comment]
@hrishikeshpaul Can we add enter key behavior to this too? For example, when enter is pressed, the first option in the list is automatically selected.

Also for #1, what do you mean by "entities"? are these the nodes in the graph, or just the items in the search list? I imaging hiding entire sets of nodes in the graph based on the search filter is not what want.

**Comment by hrishikeshpaul** [comment]
@macvogelsang the enter thing, I'm still thinking about it. There are instances where there are multiple structures that have the same name (but are different because of diff uberon IDs, eg myocardium in the heart). So by selecting the very first one could lead to problems, I feel. What do you think?

Entities mean AS,CT or B. You're not hiding any nodes in the graph. If you select the AS checkbox, you hide the CT and B names in the search dropdown.

**Comment by macvogelsang** [comment]
If the first structure in the list is not what the user intended to select, they can always manually deselect it. I think more often than not the 'enter' shortcut should help select things faster if the search term is specific enough. Plus, we could track the enter press in google analytics.

Also, we should be sure to add a task for updating the documentation to reflect all the new search functionalities. That way the above enter key behavior is more clear to the user.

**Comment by macvogelsang** [comment]
Oh, I also was going to suggest a "deselect all" button to quickly clear all selected structures. Would that be useful? @hrishikeshpaul

**Comment by hrishikeshpaul** [comment]
> If the first structure in the list is not what the user intended to select, they can always manually deselect it. I think more often than not the 'enter' shortcut should help select things faster if the search term is specific enough. Plus, we could track the enter press in google analytics.

Also, we should be sure to add a task for updating the documentation to reflect all the new search functionalities. That way the above enter key behavior is more clear to the user.

Alright you can add the enter functionality.

> Also, we should be sure to add a task for updating the documentation to reflect all the new search functionalities. That way the above enter key behavior is more clear to the user.

Yep, I'll add the task once phase 2 is done. Currently it should work the same as phase 1.

> Oh, I also was going to suggest a "deselect all" button to quickly clear all selected structures. Would that be useful?

You're right, that would be super helpful. Add this aswell!

**Comment by macvogelsang** [comment]
> Yep, I'll add the task once phase 2 is done. Currently it should work the same as phase 1.

I meant it should be included in the phase 2 work. Docs should be updated as soon as the phase 2 features are complete, I don't think it needs to be a separate PR.

**Comment by hrishikeshpaul** [comment]
I see, alright, do you want to do that too?

**Comment by macvogelsang** [comment]
Yeah, I'm totally fine doing that. Would be good for me to understand how the documentation works better.

**Comment by hrishikeshpaul** [comment]
Alright, I'll add that too

**Comment by hrishikeshpaul** [comment]
@macvogelsang see if you can incorporate #39 into this aswell
